src/core/instance/state.js

```diff
@@ -148,8 +148,14 @@ function getData (data: Function, vm: Component): any {
 const computedWatcherOptions = { lazy: true }
 
 function initComputed (vm: Component, computed: Object) {
+  if (!isPlainObject(computed)) {
```

Comment:
You should refactor this block into a function and add the check there too so it is stripped when bundling:

`process.env.NODE_ENV !== 'production' && checkOptionType(vm, 'computed')`